### PR TITLE
Protocol for WLToys V911S

### PR DESCRIPTION
**Pull request by goebish**
Add WLToys V911S protocol

Channel 5: calibrate

Forum: https://www.deviationtx.com/forum/protocol-development/8043-protocol-for-wl-heli-v911-s

### CODE REVIEW

File: src/protocol/v911s_nrf24l01.c [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  //Allows the linker to properly relocate
+  #define V911S_Cmds PROTO_Cmds
+  #pragma long_calls
+#endif
+#include "common.h"
+#include "interface.h"
+#include "mixer.h"
+#include "config/model.h"
+#include "config/tx.h" // for Transmitter
+#include "music.h"
```
Comment (howard0su) [inline src/protocol/v911s_nrf24l01.c]: you didn't reference music functions. I am also thinking to cleanup all existing reference to Music module in protocols, which doesn't really make sense to have protocol to control the user experience.

File: src/protocol/v911s_nrf24l01.c [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        tx_power = Model.tx_power;
+        NRF24L01_SetPower(tx_power);
+    }
+}
+
+static void V911S_init()
+{
+    NRF24L01_Initialize();
+    NRF24L01_SetTxRxMode(TX_EN);
+    XN297_SetTXAddr((u8 *)"\x4B\x4E\x42\x4E\x44", 5);          // Bind address
```
Comment (howard0su) [inline src/protocol/v911s_nrf24l01.c]: Does this prevent us having two people control two planes in the same model?

File: src/protocol/v911s_nrf24l01.c [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    NRF24L01_WriteReg(NRF24L01_05_RF_CH, V911S_RF_BIND_CHANNEL);    // Bind channel
+    NRF24L01_FlushTx();
+    NRF24L01_FlushRx();
+    NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);    // Clear data ready, data sent, and retransmit
+    NRF24L01_WriteReg(NRF24L01_01_EN_AA, 0x00);     // No Auto Acknowldgement on all data pipes
+    NRF24L01_WriteReg(NRF24L01_02_EN_RXADDR, 0x01); // Enable data pipe 0 only
+    NRF24L01_SetBitrate(NRF24L01_BR_250K);          // 250Kbps
+    NRF24L01_SetPower(tx_power);
```
Comment (howard0su) [inline src/protocol/v911s_nrf24l01.c]: set tx_power variable.

File: src/protocol/v911s_nrf24l01.c (hunk 2 above)
Comment (goebish) [reply src/protocol/v911s_nrf24l01.c]: no, that's only the bind address, it's common to every transmitters.

File: src/protocol/v911s_nrf24l01.c (hunk 3 above)
Comment (goebish) [reply src/protocol/v911s_nrf24l01.c]: it's set in the initialize(u8) function

File: src/protocol/v911s_nrf24l01.c (hunk 1 above)
Comment (goebish) [reply src/protocol/v911s_nrf24l01.c]: A few protocols need audio functions, I agree it's useless here.

**Pull request merged**